### ISSUE TITLE
Extends clause under protected scope

### ISSUE THREAD

**Issue by sbaetzing** [OpenModelica/OpenModelica#8510]
### Description

Currently it is not possible to use extends clauses in protected scope. This should set scope from all elements to protected but its original scope is kept (See modelica specification section 7.1.2 "Inheritance of Protected and Public Elements").

Actual Result :
Original scope is kept

### Expected Behavior

All inherited elements should be protected

### Version and OS

 - OpenModelica Version: OpenModelica v1.19.0-dev-531-g72aca4f581 (64-bit)
 - OS:  Windows 10, 64 bit

### Additional Context

Minimal example attached
[ProtectedExtend.zip](https://github.com/OpenModelica/OpenModelica/files/8022090/ProtectedExtend.zip)

**Comment by perost** [comment]
If I flatten your example I get:
```Modelica
class ProtectedExtend.Test
  parameter Real publicExtends.a = 1.0;
  protected parameter Real publicExtends.b = 2.0;
  protected parameter Real protectedExtends.a = 1.0;
  protected parameter Real protectedExtends.b = 2.0;
end ProtectedExtend.Test;
```
Is this not correct?

**Comment by sbaetzing** [comment]
That's correct, but why parameter a (protectedExtends.a) is visible on parameter dialog and modification is valid

**Comment by perost** [comment]
> That's correct, but why parameter a (protectedExtends.a) is visible on parameter dialog and modification is valid

There's are two issues here. What you're seeing is that OMEdit doesn't know that an element is protected if it became protected due to an extends clause. This is the old issue that the API only operates on declarations and not the instance hierarchy, see #8425.

The other issue is that the compiler doesn't actually forbid modifying protected elements. This might be solved by #8512, but that will only affect the compilation of models and not interacting with models in OMEdit.

**Comment by sbaetzing** [comment]
ok thanks so i will go with the given tickets to follow this issue.
